### PR TITLE
March update; Azure Sentinel, Azure Policy, image updates; nuget updates

### PR DESCRIPTION
**Pull request by givenscj**

### CODE REVIEW

File: Hands-on lab/Before the HOL - Azure security, privacy and compliance.md [hunk 1]
```diff
@@ -77,11 +77,11 @@ Synopsis: In this exercise, you will set up your environment for use in the rest
 
 4.  Type a resource group name, such as **azsecurity-\[your initials or first name\]**.
 
-5.  Select **Create**.
+5.  Select **Review + Create**, then select **Create**
 
 6.  Select **Refresh** to see your new resource group displayed and select it.
 
-7.  Select **Automation Script**, and then select **Deploy**.
+7.  Select **Export template**, and then select **Deploy**.
```
Comment (stevekmicrosoft) [inline Hands-on lab/Before the HOL - Azure security, privacy and compliance.md]: Updated image attached that matches the instructions.
img-166423

**Comment by rkuehfus** [comment]
For the Whiteboard design session student guide I noticed we do not reference the Azure Firewall under Design a proof of concept solution or in references.  

https://docs.microsoft.com/en-us/azure/firewall/overview

There are many features here that would improve the security posture of this scenario.

Features
Azure Firewall offers the following features:
Built-in high availability
High availability is built in, so no additional load balancers are required and there is nothing you need to configure.
Unrestricted cloud scalability
Azure Firewall can scale up as much as you need to accommodate changing network traffic flows, so you don't need to budget for your peak traffic.
Application FQDN filtering rules
You can limit outbound HTTP/S traffic to a specified list of fully qualified domain names (FQDN) including wild cards. This feature does not require SSL termination.
Network traffic filtering rules
You can centrally create allow or deny network filtering rules by source and destination IP address, port, and protocol. Azure Firewall is fully stateful, so it can distinguish legitimate packets for different types of connections. Rules are enforced and logged across multiple subscriptions and virtual networks.
FQDN tags
FQDN tags make it easy for you to allow well known Azure service network traffic through your firewall. For example, say you want to allow Windows Update network traffic through your firewall. You create an application rule and include the Windows Update tag. Now network traffic from Windows Update can flow through your firewall.
Service tags
A service tag represents a group of IP address prefixes to help minimize complexity for security rule creation. You cannot create your own service tag, nor specify which IP addresses are included within a tag. Microsoft manages the address prefixes encompassed by the service tag, and automatically updates the service tag as addresses change.
Threat intelligence
Threat intelligence-based filtering can be enabled for your firewall to alert and deny traffic from/to known malicious IP addresses and domains. The IP addresses and domains are sourced from the Microsoft Threat Intelligence feed.
Outbound SNAT support
All outbound virtual network traffic IP addresses are translated to the Azure Firewall public IP (Source Network Address Translation). You can identify and allow traffic originating from your virtual network to remote Internet destinations.
Inbound DNAT support
Inbound network traffic to your firewall public IP address is translated (Destination Network Address Translation) and filtered to the private IP addresses on your virtual networks.
Azure Monitor logging
All events are integrated with Azure Monitor, allowing you to archive logs to a storage account, stream events to your Event Hub, or send them to Azure Monitor logs.

**Comment by givenscj** [comment]
Hi Robert,

This is interesting.  I'm not opposed to this, we do have a WAF-based scenario in another MCW https://github.com/Microsoft/MCW-Securing-PaaS.  Have you seen that one?  

I could see adding Azure Firewall somewhere.  Do you think that should be here instead of the other one?

Chris

**Comment by rkuehfus** [comment]
Well it says in the learning objectives - design an end-to-end solution that leverages many of Microsoft Azure’s security features.  I would consider Azure Firewall and DDoS protection to both be in play in a security design.  

Also, DDoS protection could play into Customer need #17 - When specific security events are detected (such as a port scan), allow for the execution of actions to remediate, start the investigative process or prevent further information leakage or damage.

**Comment by rkuehfus** [comment]
Small thing - for the trainer and student guide you should update the copyright 

© 2018 Microsoft Corporation. All rights reserved.

**Comment by rkuehfus** [comment]
Trainers Guide, Preferred solution, Security sensitive data, Web Tier

It says - but not the corporate website (because Azure Web Apps does not offer any configuration that provides private, virtual network only endpoints---at least until App Service Environments leave preview).

Why would using a ILB with a WAF device in ASE not bee a valid option.  I don't think its in preview.  Maybe I'm missing something.  

Couple an ILB ASE with a WAF device
Azure App Service provides many security measures that protect the system. They also help to determine whether an app was hacked. The best protection for a web application is to couple a hosting platform, such as Azure App Service, with a web application firewall (WAF). Because the ILB ASE has a network-isolated application endpoint, it's appropriate for such a use.

To learn more about how to configure your ILB ASE with a WAF device, see Configure a web application firewall with your App Service environment. This article shows how to use a Barracuda virtual appliance with your ASE. Another option is to use Azure Application Gateway. Application Gateway uses the OWASP core rules to secure any applications placed behind it. For more information about Application Gateway, see Introduction to the Azure web application firewall.

**Comment by rkuehfus** [comment]
Trainers Guide, Preferred solution, Security sensitive data, Network, Internal and External Communications

It says - the website VM's are placed within a "front-end" subnet within the Virtual Network, and the SQL Database VMs are placed in a "back-end" subnet. This separation helps to create a security boundary that can be leveraged by firewalls and intrusion detection systems to more easily detect suspicious traffic, and to quickly quarantine traffic from the websites front-end to the back-end database should a breach be detected.

I would add a subnet called public for the inbound traffic before sending the packets to the "frond-end" subnet.  This way you can use our AppGW WAF to decrypted and inspected the traffic before allowing it to pass into the subnet with the web servers.  You can have the AppGW re-encrypt the data before sending to the Web servers and provide the LB.  

https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/n-tier/n-tier-sql-server

**Comment by rkuehfus** [comment]
Trainers Guide, Preferred solution,Ensuring availability and business continuity

Describe how you would ensure that the following resources would be available in the unlikely event of an attack or intentional or unintentional data loss?
Virutal Machines
a. Azure Recovery Service Vault was utilized to make backups of virtual machines

change to 

a. Azure Recovery Service Vault **configured to use GRS** was utilized to make backups of virtual machines

**Comment by givenscj** [comment]
Thanks for all the comments.  Just keep in mind the original intent of this workshop was "privacy and compliance" and not directly focused on infrastructure aspects.  The PaaS one has a lot of these topics covered in the labs, but I can see where we have a gap between development; privacy and compliance; and infrastructure.  Lots of topics, lots of scenarios.  I'll do my best to map where I can, but understand it will be virtually impossible to integrate everything into a single MCW.

**Comment by rkuehfus** [comment]
Should we still be leading with Cloudyn vs Azure Cost Management in the section of the Trainers guide called Checklist of preferred objection handling.  We should be recommending Azure Cost Management and maybe mention we also have Cloudyn for certain scenarios we have not implemented in ACM (CSP support is coming but not there yet for example)

**Comment by ChongRX78** [comment]
> Trainers Guide, Preferred solution, Security sensitive data, Network, Internal and External Communications
> It says - the website VM's are placed within a "front-end" subnet within the Virtual Network, and the SQL Database VMs are placed in a "back-end" subnet. This separation helps to create a security boundary that can be leveraged by firewalls and intrusion detection systems to more easily detect suspicious traffic, and to quickly quarantine traffic from the websites front-end to the back-end database should a breach be detected.
> I would add a subnet called public for the inbound traffic before sending the packets to the "frond-end" subnet.  This way you can use our AppGW WAF to decrypted and inspected the traffic before allowing it to pass into the subnet with the web servers.  You can have the AppGW re-encrypt the data before sending to the Web servers and provide the LB.
> https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/n-tier/n-tier-sql-server

Wouldn't the AppGW's frontend = the public IP address? Can we not achieve what you have describe without the public subnet?

**Comment by rkuehfus** [comment]
> Thanks for all the comments.  Just keep in mind the original intent of this workshop was "privacy and compliance" and not directly focused on infrastructure aspects.  The PaaS one has a lot of these topics covered in the labs, but I can see where we have a gap between development; privacy and compliance; and infrastructure.  Lots of topics, lots of scenarios.  I'll do my best to map where I can, but understand it will be virtually impossible to integrate everything into a single MCW.

No problem, just incorporate what you think is best.  It's a huge topic to cover and we just have to be careful not to give the impression we are not following our own best practices.  Maybe say a fully Secure implementation is our of scope for this workshop?

**Comment by ChongRX78** [comment]
I agree on including Azure Firewall -- especially looking ahead in terms of how we market our security solutions. DDoS could be simply a suggested option (or just mention) in the design as the basic tier is included. There is a diagram on this page to compare basic and standard: https://docs.microsoft.com/en-us/azure/virtual-network/ddos-protection-overview 

Thank you @rkuehfus for pointing this out!

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 1, Task 1, d:

Type-o: Select Stanard should say Standard

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 1, 
Should we also ask them to check to make sure auto provision is turned on? Under Data Collection

Auto Provisioning
This enables the automatic installation of the Microsoft Monitoring Agent on all the VMs in your subscription. If enabled, any new or existing VM without an installed Microsoft Monitoring agent (MMA) extension, will have it provisioned.

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 1, Task 1,
On 5. You should now see the states change to Resolved.

I believe this take a few minutes for the state to change to Resolved.  Maybe the next time the policy is evaluated?  .

**Comment by rkuehfus** [comment]
> Trainers Guide, Preferred solution, Security sensitive data, Network, Internal and External Communications
> It says - the website VM's are placed within a "front-end" subnet within the Virtual Network, and the SQL Database VMs are placed in a "back-end" subnet. This separation helps to create a security boundary that can be leveraged by firewalls and intrusion detection systems to more easily detect suspicious traffic, and to quickly quarantine traffic from the websites front-end to the back-end database should a breach be detected.
> I would add a subnet called public for the inbound traffic before sending the packets to the "frond-end" subnet.  This way you can use our AppGW WAF to decrypted and inspected the traffic before allowing it to pass into the subnet with the web servers.  You can have the AppGW re-encrypt the data before sending to the Web servers and provide the LB.
> https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/n-tier/n-tier-sql-server
> 
> Wouldn't the AppGW's frontend = the public IP address? Can we not achieve what you have describe without the public subnet?

Generally speaking from a security best practices we recommend inbound traffic is contained in its own subnet or vnet so the traffic can be inspected before being passed on.  this subnet should be locked down with an nsg and routing should be limited.  Check out - https://docs.microsoft.com/en-us/azure/best-practices-network-security  Example 3 is the most common design I run across.   

In the end, The number and type of boundaries needed vary based on a company’s risk tolerance and the specific scenario being implemented. This decision is often made together with multiple groups within an organization, often including a risk and compliance team, a network and platform team, and an application development team. People with knowledge of security, the data involved, and the technologies being used should have a say in this decision to ensure the appropriate security stance for each implementation.

For our workshop, I'm sure you can leave it out.

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 1, Task 2, step 4

If you run into an error when running the app one thing you can do is check the NuGet packages for the solution and make sure you gave everything updated.

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 3, Task 4, step 3

Remove the ending periods on sub step b and c

In the new window that opens, run the following commands:
a. Install-Package Microsoft.CodeDom.Providers.DotNetCompilerPlatform
b. Install-Package Microsoft.IdentityModel.Clients.ActiveDirectory -Version 2.16.204221202**.**
c. Install-Package Microsoft.Azure.KeyVault**.**

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 3, Task 4, step 4 

**This step is broken.  I do not see any new entries in my web.config for KeyVault**

From Solution Explorer, double-click the web.config file to open it.
Notice the appSettings section has some token values:

Replace the ClientId and ClientSecret with the values from Task 2.
Replace the SecretUri with the Azure Key Vault secret key Uri from Task 1.
Save the file.

**This step is broken i do not see any of the code the step is talking about in the global.asax.cs file**  
Task 5: Test the solution
Open the web.config, and comment out or delete the connectionString from the file at line 78.
Open the global.asax.cs file, and place a break point at line 28.
Note: This code makes a call to get an accessToken as the application you set up above, then make a call to the Azure Key Vault using that accessToken.
Run the solution, and press F5.
You should see that you execute a call to Azure Key Vault and get back the secret (which in this case is the connection string to the Azure Database).

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 4, Task 2, step 5

Says OMS and should say Log Analytics 
Select OK, wait for the OMS workspace to be created.

Hands-on lab step-by-step, Exercise 5, Task 2, step 4

The note refers to OMS when it should say Log Analytics

Note: If you were quick going through the labs, then you may not have log data in the **OMS** workspace just yet that corresponds to "AzureMetric". You may need to wait 15-30 minutes before a query will execute.

**Comment by rkuehfus** [comment]
Hands-on lab step-by-step, Exercise 4, Task 2, step 11

It took a long time for the alert to trigger in Sentinel.  I had to run the script 3x and then still wait over 60 mins before it showed up.  Can I suggest we change the Threshold to 25?

**Comment by DawnmarieDesJardins** [comment]
@givenscj @ChongRX78 
Hi Chris,
We have lead sign-off to complete this workshop with the feedback given above.  The new name for the workshop is: **Security baseline on Azure**. Please update all documentation with the new title.
Thanks, Dawnmarie

File: Hands-on lab/Before the HOL - Security baseline on Azure.md [hunk 2]
```diff
@@ -103,6 +103,8 @@ Synopsis: In this exercise, you will set up your environment for use in the rest
 
     -  Fill in the remaining parameters, but if you change anything, be sure to note it for future reference throughout the lab.
 
+    -  The **userObjectId** can be retreived by navigating to Azure Active Directory blade and searching for your user account.  On the user account page, you will find your object id which you can copy and paste into the field.
```
Comment (stevekmicrosoft) [inline Hands-on lab/Before the HOL - Security baseline on Azure.md]: The userObjectId is not needed at this step. This instruction may be confusing at this point in the instructions,

File: Hands-on lab/HOL step-by step - Security baseline on Azure.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
         - [Task 1: Create an Azure Key Vault secret](#task-1-create-an-azure-key-vault-secret)
         - [Task 2: Create an Azure Active Directory application](#task-2-create-an-azure-active-directory-application)
         - [Task 3: Assign Azure Active Directory application permissions](#task-3-assign-azure-active-directory-application-permissions)
-        - [Task 4: Install or verify Nuget Package](#task-4-install-or-verify-nuget-package)
+        - [Task 4: Install or verify NuGet Package](#task-4-install-or-verify-nuget-package)
         - [Task 5: Test the solution](#task-5-test-the-solution)
     - [Exercise 4: Securing the network](#exercise-4-securing-the-network)
         - [Task 1: Test network security group rules \#1](#task-1-test-network-security-group-rules-\1)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Security baseline on Azure.md]: This hot link does not work

**Pull request merged**